internal/testgen/gen_suite_method_signature.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"text/template"
+
+	"github.com/Antonboom/testifylint/internal/checkers"
+)
+
+type SuiteMethodSignatureTestsGenerator struct{}
+
+func (SuiteMethodSignatureTestsGenerator) Checker() checkers.Checker {
+	return checkers.NewSuiteMethodSignature()
+}
+
+func (g SuiteMethodSignatureTestsGenerator) TemplateData() any {
+	name := g.Checker().Name()
+
+	return struct {
+		CheckerName               CheckerName
+		ReportMethodShouldBeClean string
+		ReportConflictWithIface   string
+	}{
+		CheckerName:               CheckerName(name),
+		ReportMethodShouldBeClean: QuoteReport(name + ": test method should not have any arguments or returning values"),
+		ReportConflictWithIface:   QuoteReport(name + ": method conflicts with suite.%s interface"),
+	}
+}
+
+func (SuiteMethodSignatureTestsGenerator) ErroredTemplate() Executor {
+	return template.Must(template.New("SuiteMethodSignatureTestsGenerator.ErroredTemplate").
+		Funcs(fm).
+		Parse(SuiteMethodSignatureTestTmpl))
+}
+
+func (SuiteMethodSignatureTestsGenerator) GoldenTemplate() Executor {
+	// NOTE(a.telyshev): Autofix may result in uncompiled code.
+	return nil
+}
+
+const SuiteMethodSignatureTestTmpl = header + `
+
+package {{ .CheckerName.AsPkgName }}
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
```

Comment:
See here, you are importing testify here

https://github.com/Antonboom/testifylint/pull/228#discussion_r1964238226

Reply:
Yes, this import will be appear in tests (`testdata`) and do not affect linter module.
Moreover, of course, tests should use the real testify.